src/indicators/cholera/run.R

```diff
@@ -0,0 +1,192 @@
+library(tidyverse)
+library(readxl)
+library(janitor)
+library(countrycode)
+library(zoo)
+
+source(
+  file.path(
+    "src",
+    "utils",
+    "googledrive.R"
+  )
+)
+
+source(
+  file.path(
+    "src",
+    "utils",
+    "get_country_names.R"
+  )
+)
+
+# alert helper function
+# only generate alert when boundary crossed from below
+lim_alert <- function(x, lim) {
+  x >= lim & lag(x) < lim
+}
+
+##############################
+#### DRIVE AND LOCAL DATA ####
+##############################
+
+# authorize and prep
+
+# get drive files
+drive_raw_cholera <- get_drive_file("raw_cholera")
+drive_wrangled_cholera <- get_drive_file("wrangled_cholera")
+drive_flags_cholera <- get_drive_file("flags_cholera")
+
+# temp paths for local saving
+local_raw_cholera <- tempfile(fileext = ".csv")
+local_wrangled_cholera <- tempfile(fileext = ".csv")
+local_flags_cholera <- tempfile(fileext = ".csv")
+
+#################
+#### CHOLERA ####
+#################
+
+# raw data
+df_cholera_raw <- read_csv(
+  url(
+    Sys.getenv("CERF_CHOLERA_DATA")
+  )
+)
+
+# wrangled data
+df_cholera_wrangled <- df_cholera_raw %>%
+  clean_names() %>%
+  filter(
+    event == "Cholera"
+  ) %>%
+  transmute(
+    iso3 = countryname(country, destination = "iso3c"),
+    start_date = lubridate::dmy(
+      case_when(
+        !is.na(start_of_reporting_period) ~ start_of_reporting_period,
+        !is.na(start_of_reporting_period_2) ~ start_of_reporting_period_2,
+        !is.na(start_of_reporting_period_3) ~ start_of_reporting_period_3
+      )
+    ),
+    date = week_date,
+    cholera_cases = total_cases
+  ) %>%
+  group_by(iso3) %>%
+  arrange(
+    start_date,
+    date,
+    .by_group = TRUE
+  ) %>%
+  mutate(
+    cholera_cases = na.approx(cholera_cases)
+  ) %>%
+  ungroup()
+
+# flagged data
+
+df_cholera_flags <- df_cholera_wrangled %>%
+  group_by(iso3) %>%
+  mutate(
+    flag_1k = lim_alert(cholera_cases, 1000),
+    flag_5k = lim_alert(cholera_cases, 5000),
+    flag_10k = lim_alert(cholera_cases, 10000),
+    flag_stop = cholera_cases < 1000 & lag(cholera_cases, default = 0) >= 1000,
```

Comment:
Good catch!! I think actually the simplest thing to do, which also addresses your comments on the code, is just to drop `flag_stop` and `group`. However, the general issue is that I think the data is just wrong, `start_date` for that Nigeria example you highlighted is basically incorrect. I don't want to go too deep into fixing what is essentially an error from elsewhere.

In the end, I think the code is much cleaner without trying to capture this error. What we can do is flag it to CERF and see if they want to deal with it their side on the scrapers, because they are also managing the raw data scrapers. Anyway, since this was a couple of years ago, I hope the error won't happen again. I have checked the data, and we have no other instances of this occurring. There are some times when cases decrease slightly in a following week, but never permanently and have no impact on flagging.

Reply:
So yeah, no instances of caseloads "fluctuating" around 1,000 or so cases thankfully. These are supposed to be cumulative as you noted, it's just a very rough way of getting them.